tests/unit/common/browserInfoHelper.test.js

```diff
@@ -221,6 +223,39 @@ describe("BrowserInfoHelper", () => {
     });
   });
 
+  describe("getDeviceLocalIPTimeStamp", () => {
+    beforeEach(() => {
+      resetDeviceIpTimeStamp();
+    });
+
+    afterEach(() => {
+      jest.restoreAllMocks();
+      resetDeviceIpTimeStamp();
+    });
+
+    it("getDeviceLocalIPTimeStamp throws NO_IP_TIMESTAMP_FOUND for empty timestamp string", async () => {
+      await expect(async() => getDeviceLocalIPTimeStamp())
+      .rejects
+      .toBeErrorWithMessage('NO_IP_TIMESTAMP_FOUND');  
+    });
+
+    it("getDeviceLocalIPTimeStamp returns timestamp after IP has been retrieved", async () => {
+      const mockTimeStamp = "2021-06-03T13:02:22.107Z"
+      global.RTCPeerConnection = MockRTCPeerConnection;
+      global.Date = class DateMock {
+        constructor() {
+        }
+        toISOString() {
+            return mockTimeStamp;
+        }
+    };
+      const dateSpy = jest.spyOn(global.Date.prototype, 'toISOString');
+      await getDeviceLocalIPAsString();
```

Comment:
If we do the change proposed by Reuben, we don't need to call this method before calling the one that will be returning the timestamp.